packages/babel-plugin-emotion/src/index.js

```diff
@@ -94,9 +94,26 @@ export function replaceCssWithCallExpression(
   }
 }
 
+function getFilepathFromState(state) {
+  const path = state.file.opts.filename;
+
+  return process.env.NODE_ENV === 'test' ? basename(path) : path;
+}
+
 export function buildStyledCallExpression(identifier, tag, path, state, t) {
   const identifierName = getIdentifierName(path, t)
 
+  let stableClassName = `el-${hashString(getFilepathFromState(state))}`
+
+  if (identifierName) {
+    stableClassName += `-${identifierName}`
+  }
+
+  const targetExpression = t.objectProperty(
+    t.identifier('target'),
+    t.stringLiteral(stableClassName)
+  )
+
```

Comment:
Could you use this code to generate the class name? (especially with the `css-` prefix)

https://github.com/emotion-js/emotion/blob/54f054a51e72ade3673f3d5a8fd0b0bab6e35643/packages/babel-plugin-emotion/src/index.js#L81-L134

Reply:
Happy to change `el-` to `css-`. Why the more complicated hash function though? styled-components has a lot of bugs around hash generation... I'd rather go with as simple a solution as possible to avoid them and noticed styled uses a lot similar code to what was removed.

Reply:
The more complicated hash function is required so that it's unique, deterministic and won't change even if the project has a different full path (hence using the path to the nearest package.json).

Reply:
What bugs around hash generation are you referring to?

Reply:
https://github.com/styled-components/babel-plugin-styled-components/issues/66 Is the big one

Reply:
Backing up to package JSON definitely makes sense though. I’ll implement that.

Reply:
@probablyup styled-components/babel-plugin-styled-components#66 would already happen with your code since you're using the `identifierName` in the hash.

Reply:
Thinking about it more, using the identifierName is the real problem in the hash so using the module name, path and the original code will work fine and styled-components/babel-plugin-styled-components#66 won't occur.

Reply:
The old code appears to use identifierName as well: https://github.com/emotion-js/emotion/blob/54f054a51e72ade3673f3d5a8fd0b0bab6e35643/packages/babel-plugin-emotion/src/index.js#L162

Reply:
I'm purely referring to the code above where the identifier name is not used.

Reply:
Would tracking position not cause the same issue if babel is rewriting the variable assignment and potentially introducing more lines?

Reply:
Nvm I think I get it now. It's using an incrementor to just refer vaguely to the nth component in the file, right?

Reply:
@probablyup yes.